sourcegraph/sourcegraph
# Add frontend component to allow hard deleting an organization

Jira Ticket: CLOUD-81

This PR adds the frontend changes to allow Cloud users that are org members to hard delete the organizations they belong to.

## Test plan

For manual tests

1) When **not in DotCom mode**, the new component should not be visible and the current flow for deleting an org won't change
- Run `sg start` , create an org and visit the org Settings Page. 
- The "Delete this Organization" component should not be visible
- Go to the Site Admin area >  Organization and delete the new organization.
-  Check the database to confirm that the org has been only soft deleted but not hard deleted.
 
2) When in DotCom mode, the new component should be visible and orgs can be hard deleted by org members
- Run `sg start dotcom` , create an org and visit the org Settings Page. 
- The "Delete this Organization" component should be visible.
-  Delete the new organization. You should be redirected to the User Settings page.
-  Check the database to confirm that the org has been hard deleted.

Videos showing case 1 and 2:
https://user-images.githubusercontent.com/1552863/156676333-286463ef-aa4d-4eb8-8e7d-6c80ba66970c.mov
https://user-images.githubusercontent.com/1552863/156676371-b36058a6-e334-4b09-9f7e-705ba90aa69b.mov

Images:
<img width="1166" alt="Screen Shot 2022-03-03 at 4 20 10 PM" src="https://user-images.githubusercontent.com/1552863/156676516-7ca030fc-701a-43e9-8efa-f25a23624a2c.png">
<img width="1338" alt="Screen Shot 2022-03-03 at 4 20 33 PM" src="https://user-images.githubusercontent.com/1552863/156676542-0aa12357-4cf2-474c-b8a2-4653b9fe9609.png">
